sympy/solvers/solvers.py

```diff
@@ -1136,6 +1136,25 @@ def _has_piecewise(e):
         if _has_piecewise(fi):
             f[i] = piecewise_fold(fi)
 
+    # expand double angles; in general, expand_trig will allow
+    # more roots to be found but this is not a great solultion
+    # to not returning a parametric solution, otherwise
+    # many values can be returned that have a simple
+    # relationship between values
+    for i, fi in enumerate(f):
+        for t in fi.atoms(TrigonometricFunction):
+            if not fi.coeff(t).is_Number:
+                continue
+            # linear shift
+            if t.args[0].is_Add and len(t.args[0].args) == 2:
+                ind, dep = t.args[0].as_independent(*symbols)
+                if dep in symbols or -dep in symbols:
+                    fi = fi.xreplace({t: t.expand(trig=True)})
```

Comment:
Will this not expand something like `cos(1000*x + 1000*a)` (if `a` is not in symbols)?

Reply:
In this case `dep` will be `1000x`:
```python
>>> 1000*x+1000*y
1000*x + 1000*y
>>> _.as_independent(x)
(1000*y, 1000*x)
```